tldw_Server_API/app/api/v1/endpoints/character_chat_sessions.py

```
@@ -534,6 +534,145 @@ def _validate_and_truncate_tool_calls(tool_calls: Any) -> Optional[list]:
+    speaker_name: str,
+) -> tuple[str, dict[str, Any]] | None:
+    """Return an existing persisted assistant message by stable message id."""
+    message = db.get_message_by_id(assistant_message_id)
+    if not message:
+        return None
+    if str(message.get("conversation_id") or "").strip() != str(chat_id).strip():
+        raise ConflictError("assistant_message_id already exists in another conversation")
+    if message.get("deleted"):
+        raise ConflictError("assistant_message_id references a deleted message")
+    if str(message.get("sender") or "") != speaker_name:
+        raise ConflictError("assistant_message_id already exists for a different speaker")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)